### PR TITLE
Add build info for multi-arch (i386, armel, armv6, armv7hf)

### PR DESCRIPTION
**Pull request by nghiant2710**
This change is [Reviewable](https://reviewable.io/reviews/resin-io/resin-supervisor/195)

### CODE REVIEW

File: base-image/build/conf/machine/generic-armv5.conf [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#@TYPE: Machine
+#@NAME: Generic armv5te machine
+#@DESCRIPTION: Machine configuration for armv5te based boards
+
+require conf/machine/include/qemu.inc
+require conf/machine/include/arm/arch-armv5.inc
+
+KERNEL_IMAGETYPE = "zImage"
+
+SERIAL_CONSOLES = "115200;ttyAMA0 115200;ttyAMA1"
```
Comment (petrosagg) [inline base-image/build/conf/machine/generic-armv5.conf]: Why don't  need any of these two variables. Are they required?

File: base-image/build/conf/machine/generic-armv5.conf [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
```
Comment (petrosagg) [inline base-image/build/conf/machine/generic-armv5.conf]: trailing whitespace

File: base-image/build/conf/machine/generic-armv6.conf [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+#@DESCRIPTION: Machine configuration for armv6 based boards
+
+require conf/machine/include/qemu.inc
+require conf/machine/include/arm/arch-armv6.inc
+
+KERNEL_IMAGETYPE = "zImage"
+
+SERIAL_CONSOLES = "115200;ttyAMA0 115200;ttyAMA1"
+
+
```
Comment (petrosagg) [inline base-image/build/conf/machine/generic-armv6.conf]: trailing whitespace

File: base-image/build/conf/machine/generic-armv5.conf [hunk 4]
```diff
@@ -0,0 +1,11 @@
+#@TYPE: Machine
+#@NAME: Generic armv5te machine
+#@DESCRIPTION: Machine configuration for armv5te based boards
+
+require conf/machine/include/qemu.inc
```
Comment (petrosagg) [inline base-image/build/conf/machine/generic-armv5.conf]: We don't need QEMU either

File: base-image/automation/jenkins-build.sh [hunk 5]
```diff
@@ -5,21 +5,43 @@ set -o pipefail
...
-else
-	echo "rootfs is missing!"
-	exit 1
-fi
+for machine in $MACHINE_LIST; do
+	case "$device" in
+	'qemux86-64')
+		REPO='resin/amd64-supervisor-base'
+	;;
+	'qemux86')
```
Comment (petrosagg) [inline base-image/automation/jenkins-build.sh]: The default qemu machines set some properties that are unrelated to containers. Let's create generic machine definitions for the qemux86 and qemux86-64 machines too.

`generic-x86` should only include `conf/machine/include/tune-i586.inc` and `generic-x86-64` should only include `conf/machine/include/tune-core2.inc` and define `DEFAULTTUNE ?= "core2-64".

File: base-image/build.sh [hunk 6]
```diff
@@ -8,6 +8,6 @@ mkdir -p $BUILD_DIR
 cp -r $SOURCE_DIR/* $BUILD_DIR/
 cd $BUILD_DIR
 source poky/oe-init-build-env build
+sed -e s~#{MACHINE_NAME}~"$TARGET_MACHINE"~g conf/local.conf.tpl > conf/local.conf
```
Comment (petrosagg) [inline base-image/build.sh]: No need to do this. you can invoke bitbake like so: `MACHINE=generic-armv6 bitbake core-image-minimal`

File: base-image/build/conf/local.conf.tpl [hunk 7]
```diff
@@ -38,7 +38,7 @@ PREFERRED_PROVIDER_virtual/kernel="linux-dummy"
 #MACHINE ?= "edgerouter"
 #
 # This sets the default machine to be qemux86 if no other machine is selected:
-MACHINE ??= "qemux86-64"
+MACHINE ??= "#{MACHINE_NAME}"
```
Comment (petrosagg) [inline base-image/build/conf/local.conf.tpl]: Let's keep this as a normal config. See comment above

File: base-image/build/conf/machine/generic-armv5.conf (hunk 1 above)
Comment (nghiant2710) [inline base-image/build/conf/machine/generic-armv5.conf]: Just checked, these variables are kernel variables define type of kernel and the serial consoles (TTYs) to enable using getty. So can be removed.

File: base-image/build/conf/machine/generic-armv5.conf (hunk 2 above)
Comment (nghiant2710) [inline base-image/build/conf/machine/generic-armv5.conf]: Updated

File: base-image/build/conf/machine/generic-armv5.conf (hunk 4 above)
Comment (nghiant2710) [inline base-image/build/conf/machine/generic-armv5.conf]: removed

File: base-image/build/conf/machine/generic-armv6.conf (hunk 3 above)
Comment (nghiant2710) [inline base-image/build/conf/machine/generic-armv6.conf]: removed

File: base-image/automation/jenkins-build.sh (hunk 5 above)
Comment (nghiant2710) [inline base-image/automation/jenkins-build.sh]: Created and  switched to the new generic machines

File: base-image/build.sh (hunk 6 above)
Comment (nghiant2710) [inline base-image/build.sh]: Updated

File: base-image/build/conf/local.conf.tpl (hunk 7 above)
Comment (nghiant2710) [inline base-image/build/conf/local.conf.tpl]: UpdateD
